Index: toolkit/crashreporter/client/crashreporter_linux.cpp
===================================================================
RCS file: /cvsroot/mozilla/toolkit/crashreporter/client/crashreporter_linux.cpp,v
retrieving revision 1.16
diff -d -u -p -r1.16 crashreporter_linux.cpp
--- toolkit/crashreporter/client/crashreporter_linux.cpp	7 Mar 2008 22:17:17 -0000	1.16
+++ toolkit/crashreporter/client/crashreporter_linux.cpp	8 Mar 2008 05:48:24 -0000
@@ -47,6 +47,7 @@
 
 #include <algorithm>
 #include <cctype>
+#include <climits>
 
 #include <signal.h>
 
