diff -upU8 mozilla/security/manager/Makefile.in.mozilla-build mozilla/security/manager/Makefile.in
--- mozilla/security/manager/Makefile.in.mozilla-build	2009-10-07 14:39:15.000000000 +0200
+++ mozilla/security/manager/Makefile.in	2009-10-07 14:39:54.000000000 +0200
@@ -174,17 +174,16 @@ endif
 DEFAULT_GMAKE_FLAGS = MAKE="$(MAKE) -j1" -j1
 DEFAULT_GMAKE_FLAGS += CC="$(CC)"
 DEFAULT_GMAKE_FLAGS += SOURCE_MD_DIR=$(ABS_DIST)
 DEFAULT_GMAKE_FLAGS += DIST=$(ABS_DIST)
 DEFAULT_GMAKE_FLAGS += NSPR_INCLUDE_DIR=$(NSPR_INCLUDE_DIR)
 DEFAULT_GMAKE_FLAGS += NSPR_LIB_DIR=$(NSPR_LIB_DIR)
 DEFAULT_GMAKE_FLAGS += MOZILLA_CLIENT=1
 DEFAULT_GMAKE_FLAGS += NO_MDUPDATE=1
-DEFAULT_GMAKE_FLAGS += NSS_ENABLE_ECC=1
 DEFAULT_GMAKE_FLAGS += IMPORT_LIB_SUFFIX=".$(IMPORT_LIB_SUFFIX)"
 ABS_topsrcdir   := $(shell cd $(topsrcdir); pwd)
 ifneq ($(ABS_topsrcdir),$(MOZ_BUILD_ROOT))
 DEFAULT_GMAKE_FLAGS += BUILD_TREE=$(MOZ_BUILD_ROOT)
 endif
 ifndef MOZ_DEBUG
 DEFAULT_GMAKE_FLAGS += BUILD_OPT=1 OPT_CODE_SIZE=1
 endif
diff -upU8 mozilla/security/manager/ssl/src/Makefile.in.mozilla-build mozilla/security/manager/ssl/src/Makefile.in
--- mozilla/security/manager/ssl/src/Makefile.in.mozilla-build	2009-10-07 14:39:39.000000000 +0200
+++ mozilla/security/manager/ssl/src/Makefile.in	2009-10-07 14:39:52.000000000 +0200
@@ -130,18 +130,16 @@ REQUIRES	= nspr \
 		  content \
 		  xpconnect \
 		  unicharutil \
 		  pipboot \
 		  $(NULL)
 
 EXTRA_DEPS = $(NSS_DEP_LIBS)
 
-DEFINES += -DNSS_ENABLE_ECC
-
 # Use local includes because they are inserted before INCLUDES
 # so that Mozilla's nss.h is used, not glibc's
 LOCAL_INCLUDES += $(NSS_CFLAGS)
 
 EXTRA_DSO_LDOPTS += \
 		$(MOZ_UNICHARUTIL_LIBS) \
 		$(MOZ_COMPONENT_LIBS) \
 		$(MOZ_JS_LIBS) \
