diff -up mozilla/storage/build/Makefile.in.old mozilla/storage/build/Makefile.in
--- mozilla/storage/build/Makefile.in.old	2007-08-10 18:19:57.000000000 +0200
+++ mozilla/storage/build/Makefile.in	2007-11-14 09:17:00.000000000 +0100
@@ -21,7 +21,6 @@
 #
 # Contributor(s):
 #   Vladimir Vukicevic <vladimir.vukicevic@oracle.com>
-#   Shawn Wilsher <me@shawnwilsher.com>
 #
 # Alternatively, the contents of this file may be used under the terms of
 # either of the GNU General Public License Version 2 or later (the "GPL"),
@@ -71,10 +70,9 @@ LOCAL_INCLUDES = \
 
 SHARED_LIBRARY_LIBS = \
 	../src/$(LIB_PREFIX)storage_s.$(LIB_SUFFIX) \
+	$(DEPTH)/db/sqlite3/src/$(LIB_PREFIX)sqlite3_s.$(LIB_SUFFIX) \
 	$(NULL)
 
-EXTRA_DSO_LIBS += sqlite3
-
 EXTRA_DSO_LDOPTS += \
 	$(LIBS_DIR) \
 	$(EXTRA_DSO_LIBS) \
diff -up mozilla/toolkit/library/libxul-config.mk.old mozilla/toolkit/library/libxul-config.mk
--- mozilla/toolkit/library/libxul-config.mk.old	2007-09-17 01:48:08.000000000 +0200
+++ mozilla/toolkit/library/libxul-config.mk	2007-11-14 09:14:16.000000000 +0100
@@ -21,7 +21,6 @@
 # the Initial Developer. All Rights Reserved.
 #
 # Contributor(s):
-#   Shawn Wilsher <me@shawnwilsher.com>
 #
 # Alternatively, the contents of this file may be used under the terms of
 # either of the GNU General Public License Version 2 or later (the "GPL"),
@@ -223,7 +222,6 @@ endif
 
 ifdef MOZ_STORAGE
 COMPONENT_LIBS += storagecomps
-EXTRA_DSO_LIBS += sqlite3
 endif
 
 ifdef MOZ_PLACES
diff -up mozilla/xpcom/stub/Makefile.in.old mozilla/xpcom/stub/Makefile.in
--- mozilla/xpcom/stub/Makefile.in.old	2007-10-13 23:13:52.000000000 +0200
+++ mozilla/xpcom/stub/Makefile.in	2007-11-14 09:15:28.000000000 +0100
@@ -91,7 +91,6 @@ DEPENDENT_LIBS_LIST += \
 ifdef MOZ_ENABLE_LIBXUL
 
 DEPENDENT_LIBS_LIST += \
-	$(LIB_PREFIX)sqlite3$(DLL_SUFFIX) \
 	$(LIB_PREFIX)softokn3$(DLL_SUFFIX) \
 	$(LIB_PREFIX)nss3$(DLL_SUFFIX) \
 	$(LIB_PREFIX)ssl3$(DLL_SUFFIX) \
diff -up mozilla/config/static-config.mk.old mozilla/config/static-config.mk
--- mozilla/config/static-config.mk.old	2007-07-24 00:02:20.000000000 +0200
+++ mozilla/config/static-config.mk	2007-11-14 09:14:16.000000000 +0100
@@ -98,10 +98,6 @@ ifdef MOZ_ENABLE_PANGO
 STATIC_EXTRA_LIBS	+= $(MOZ_PANGO_LIBS)
 endif
 
-ifdef MOZ_STORAGE
-EXTRA_DSO_LIBS += sqlite3
-endif
-
 ifdef MOZ_ENABLE_STARTUP_NOTIFICATION
 STATIC_EXTRA_LIBS	+= $(MOZ_STARTUP_NOTIFICATION_LIBS)
 endif
diff -up mozilla/config/system-headers.old mozilla/config/system-headers
--- mozilla/config/system-headers.old	2007-10-12 14:34:46.000000000 +0200
+++ mozilla/config/system-headers	2007-11-14 09:14:16.000000000 +0100
@@ -617,8 +617,6 @@ SIOUX.h
 size_t.h
 someincludefile.h
 Sound.h
-sqlite3.h
-sqlite3file.h
 ssdef.h
 sslerr.h
 ssl.h
diff -up mozilla/db/sqlite3/src/sqlite3file.h.old mozilla/db/sqlite3/src/sqlite3file.h
--- mozilla/db/sqlite3/src/sqlite3file.h.old	2007-07-25 04:18:54.000000000 +0200
+++ mozilla/db/sqlite3/src/sqlite3file.h	2007-11-14 09:14:16.000000000 +0100
@@ -199,11 +199,7 @@ struct sqlite3OsVtbl {
   /*
   ** Files other than os.c just reference the global virtual function table. 
   */
-#ifdef XP_WIN
-  extern __declspec(dllimport) struct sqlite3OsVtbl sqlite3Os;
-#else
   extern struct sqlite3OsVtbl sqlite3Os;
-#endif // windows symbol ifdef
 #endif /* _SQLITE_OS_C_ */
 
 } // extern "C"
diff -up mozilla/db/sqlite3/src/Makefile.in.old mozilla/db/sqlite3/src/Makefile.in
--- mozilla/db/sqlite3/src/Makefile.in.old	2007-09-16 03:47:04.000000000 +0200
+++ mozilla/db/sqlite3/src/Makefile.in	2007-11-14 09:20:17.000000000 +0100
@@ -45,15 +45,9 @@ VPATH		= @srcdir@
 include $(DEPTH)/config/autoconf.mk
 
 MODULE           = sqlite3
-LIBRARY_NAME     = sqlite3
-FORCE_SHARED_LIB = 1
-VISIBILITY_FLAGS =
-
-ifeq (,$(filter-out WINNT WINCE,$(OS_ARCH)))
-ifndef GNU_CC
-DEFFILE = $(win_srcdir)/sqlite.def
-endif
-endif
+LIBRARY_NAME     = sqlite3_s
+MODULE_NAME      = sqlite3
+FORCE_STATIC_LIB = 1
 
 ifdef GNU_CC
 MODULE_OPTIMIZE_FLAGS = -O3
@@ -83,7 +77,15 @@ DEFINES += -DOS2_HIGH_MEMORY
 endif
 endif
 
+ifdef ENABLE_TESTS
+##LIBS = $(DIST)/lib/$(LIB_PREFIX)sqlite3_s.$(LIB_SUFFIX)
+##SIMPLE_PROGRAMS = sqlite-shell
+endif
+
 include $(topsrcdir)/config/rules.mk
+    
+sqlite-shell.$(OBJ_SUFFIX): shell.c
+	$(ELOG) $(CC) $(OUTOPTION)$@ -c $(COMPILE_CFLAGS) $(_VPATH_SRCS)
 
 # next line allows use of MOZ_OBJDIR in .mozconfig with older gcc on BeOS, maybe others
 LOCAL_INCLUDES += -I$(srcdir)
