diff -up xulrunner-1.9.2.13/mozilla-1.9.2/browser/installer/package-manifest.in.513747 xulrunner-1.9.2.13/mozilla-1.9.2/browser/installer/package-manifest.in
--- xulrunner-1.9.2.13/mozilla-1.9.2/browser/installer/package-manifest.in.513747	2010-12-03 16:47:53.000000000 +0100
+++ xulrunner-1.9.2.13/mozilla-1.9.2/browser/installer/package-manifest.in	2011-01-03 13:29:14.000000000 +0100
@@ -99,7 +99,7 @@
 #elifdef XP_OS2
 @BINPATH@/plugins/npnulos2.dll
 #endif
-@BINPATH@/@DLL_PREFIX@sqlite3@DLL_SUFFIX@
+@BINPATH@/@DLL_PREFIX@mozsqlite3@DLL_SUFFIX@
 @BINPATH@/README.txt
 @BINPATH@/LICENSE
 @BINPATH@/blocklist.xml
diff -up xulrunner-1.9.2.13/mozilla-1.9.2/browser/installer/removed-files.in.513747 xulrunner-1.9.2.13/mozilla-1.9.2/browser/installer/removed-files.in
--- xulrunner-1.9.2.13/mozilla-1.9.2/browser/installer/removed-files.in.513747	2010-12-03 16:47:53.000000000 +0100
+++ xulrunner-1.9.2.13/mozilla-1.9.2/browser/installer/removed-files.in	2011-01-03 13:20:16.000000000 +0100
@@ -811,5 +811,6 @@ install.rdf
 components/nsUpdateService.js
 components/nsUpdateServiceStub.js
 #endif
+@DLL_PREFIX@sqlite3@DLL_SUFFIX@
 old-homepage-default.properties
 mozilla-runtime@BIN_SUFFIX@
diff -up xulrunner-1.9.2.13/mozilla-1.9.2/build/automation.py.in.513747 xulrunner-1.9.2.13/mozilla-1.9.2/build/automation.py.in
--- xulrunner-1.9.2.13/mozilla-1.9.2/build/automation.py.in.513747	2010-12-03 16:47:53.000000000 +0100
+++ xulrunner-1.9.2.13/mozilla-1.9.2/build/automation.py.in	2011-01-03 13:20:16.000000000 +0100
@@ -448,7 +448,7 @@ user_pref("camino.use_system_proxy_setti
     if self.UNIXISH or self.IS_MAC:
       envVar = "LD_LIBRARY_PATH"
       if self.IS_MAC:
-        envVar = "DYLD_LIBRARY_PATH"
+        envVar = "DYLD_LIBRARY_PATH_DISABLED"
       else: # unixish
         env['MOZILLA_FIVE_HOME'] = xrePath
       if envVar in env:
diff -up xulrunner-1.9.2.13/mozilla-1.9.2/configure.in.513747 xulrunner-1.9.2.13/mozilla-1.9.2/configure.in
--- xulrunner-1.9.2.13/mozilla-1.9.2/configure.in.513747	2011-01-03 13:20:16.000000000 +0100
+++ xulrunner-1.9.2.13/mozilla-1.9.2/configure.in	2011-01-03 13:28:11.000000000 +0100
@@ -6332,8 +6332,8 @@ MOZ_NATIVE_SQLITE= )
 
 if test -z "$MOZ_NATIVE_SQLITE"
 then
-    SQLITE_CFLAGS=
-    SQLITE_LIBS='$(call EXPAND_LIBNAME_PATH,sqlite3,$(DIST)/lib)'
+    SQLITE_CFLAGS=    
+    SQLITE_LIBS='$(call EXPAND_LIBNAME_PATH,mozsqlite3,$(DIST)/lib)'
 else
     PKG_CHECK_MODULES(SQLITE, sqlite3 >= $SQLITE_VERSION)
     dnl ===================================
diff -up xulrunner-1.9.2.13/mozilla-1.9.2/db/sqlite3/src/Makefile.in.513747 xulrunner-1.9.2.13/mozilla-1.9.2/db/sqlite3/src/Makefile.in
--- xulrunner-1.9.2.13/mozilla-1.9.2/db/sqlite3/src/Makefile.in.513747	2010-12-03 16:47:59.000000000 +0100
+++ xulrunner-1.9.2.13/mozilla-1.9.2/db/sqlite3/src/Makefile.in	2011-01-03 13:20:16.000000000 +0100
@@ -46,7 +46,7 @@ VPATH		= @srcdir@
 include $(DEPTH)/config/autoconf.mk
 
 MODULE           = sqlite3
-LIBRARY_NAME     = sqlite3
+LIBRARY_NAME     = mozsqlite3
 FORCE_SHARED_LIB = 1
 VISIBILITY_FLAGS =
 LIB_IS_C_ONLY    = 1
diff -up xulrunner-1.9.2.13/mozilla-1.9.2/db/sqlite3/src/sqlite.def.513747 xulrunner-1.9.2.13/mozilla-1.9.2/db/sqlite3/src/sqlite.def
--- xulrunner-1.9.2.13/mozilla-1.9.2/db/sqlite3/src/sqlite.def.513747	2010-12-03 16:47:59.000000000 +0100
+++ xulrunner-1.9.2.13/mozilla-1.9.2/db/sqlite3/src/sqlite.def	2011-01-03 13:20:16.000000000 +0100
@@ -33,7 +33,7 @@
 ;
 ; ***** END LICENSE BLOCK *****
 
-LIBRARY sqlite3.dll
+LIBRARY mozsqlite3.dll
 
 EXPORTS
         sqlite3_aggregate_context
diff -up xulrunner-1.9.2.13/mozilla-1.9.2/security/manager/Makefile.in.513747 xulrunner-1.9.2.13/mozilla-1.9.2/security/manager/Makefile.in
--- xulrunner-1.9.2.13/mozilla-1.9.2/security/manager/Makefile.in.513747	2011-01-03 13:20:16.000000000 +0100
+++ xulrunner-1.9.2.13/mozilla-1.9.2/security/manager/Makefile.in	2011-01-03 13:30:23.000000000 +0100
@@ -179,6 +179,9 @@ DEFAULT_GMAKE_FLAGS += NSPR_INCLUDE_DIR=
 DEFAULT_GMAKE_FLAGS += NSPR_LIB_DIR=$(NSPR_LIB_DIR)
 DEFAULT_GMAKE_FLAGS += MOZILLA_CLIENT=1
 DEFAULT_GMAKE_FLAGS += NO_MDUPDATE=1
+ifndef MOZ_NATIVE_SQLITE
+DEFAULT_GMAKE_FLAGS += SQLITE=$(call EXPAND_LIBNAME,mozsqlite3)
+endif
 ifdef NSS_DISABLE_DBM 
 DEFAULT_GMAKE_FLAGS += NSS_DISABLE_DBM=1
 endif
diff -up xulrunner-1.9.2.13/mozilla-1.9.2/xpcom/stub/Makefile.in.513747 xulrunner-1.9.2.13/mozilla-1.9.2/xpcom/stub/Makefile.in
--- xulrunner-1.9.2.13/mozilla-1.9.2/xpcom/stub/Makefile.in.513747	2010-12-03 16:49:21.000000000 +0100
+++ xulrunner-1.9.2.13/mozilla-1.9.2/xpcom/stub/Makefile.in	2011-01-03 13:20:16.000000000 +0100
@@ -85,7 +85,7 @@ DEPENDENT_LIBS_LIST += \
 ifdef MOZ_ENABLE_LIBXUL
 
 DEPENDENT_LIBS_LIST += \
-	$(LIB_PREFIX)sqlite3$(DLL_SUFFIX) \
+	$(LIB_PREFIX)mozsqlite3$(DLL_SUFFIX) \
 	$(LIB_PREFIX)nssutil3$(DLL_SUFFIX) \
 	$(LIB_PREFIX)softokn3$(DLL_SUFFIX) \
 	$(LIB_PREFIX)nss3$(DLL_SUFFIX) \
