--- mozilla-1.9.2/js/src/shell/js.cpp.sbrk	2010-01-06 04:35:17.000000000 +0100
+++ mozilla-1.9.2/js/src/shell/js.cpp	2010-01-13 21:47:40.000000000 +0100
@@ -1119,7 +1119,7 @@ GC(JSContext *cx, uintN argc, jsval *vp)
     char buf[256];
     JS_snprintf(buf, sizeof(buf), "before %lu, after %lu, break %08lx\n",
                 (unsigned long)preBytes, (unsigned long)rt->gcBytes,
-#ifdef HAVE_SBRK
+#if 0
                 (unsigned long)sbrk(0)
 #else
                 0
