diff -up mozilla-1.9.2/security/manager/ssl/src/nsNSSComponent.cpp.462919 mozilla-1.9.2/security/manager/ssl/src/nsNSSComponent.cpp
--- mozilla-1.9.2/security/manager/ssl/src/nsNSSComponent.cpp.462919	2010-01-21 04:31:05.000000000 +0100
+++ mozilla-1.9.2/security/manager/ssl/src/nsNSSComponent.cpp	2010-02-18 16:24:49.000000000 +0100
@@ -1573,9 +1573,15 @@ nsNSSComponent::InitializeNSS(PRBool sho
       return rv;
   #endif
 
-    rv = profilePath->GetNativePath(profileStr);
-    if (NS_FAILED(rv)) 
-      return rv;
+    const char *dbdir_override = getenv("MOZPSM_NSSDBDIR_OVERRIDE");
+    if (dbdir_override && strlen(dbdir_override)) {
+      profileStr = dbdir_override;
+    }
+    else {
+      rv = profilePath->GetNativePath(profileStr);
+      if (NS_FAILED(rv)) 
+        return rv;
+    }
 
     hashTableCerts = PL_NewHashTable( 0, certHashtable_keyHash, certHashtable_keyCompare,
       certHashtable_valueCompare, 0, 0 );
