--- mozilla/xulrunner/installer/Makefile.in.gtk	2008-01-08 09:44:23.000000000 +0100
+++ mozilla/xulrunner/installer/Makefile.in	2008-01-08 09:44:23.000000000 +0100
@@ -90,6 +90,8 @@ pkg_config_files = \
 	libxul-embedding-unstable.pc \
 	mozilla-js.pc \
 	mozilla-plugin.pc \
+	mozilla-gtkmozembed.pc \
+	mozilla-gtkmozembed-embedding.pc \
 	$(NULL)
 
 ifdef MOZ_NATIVE_NSPR
--- /dev/null	2008-01-08 09:03:16.107043921 +0100
+++ mozilla/xulrunner/installer/mozilla-gtkmozembed.pc.in	2008-01-08 09:45:01.000000000 +0100
@@ -0,0 +1,8 @@
+prefix=%prefix%
+sdkdir=%sdkdir%
+includedir=%includedir%
+
+Name: mozilla-gtkembedmoz
+Description: Mozilla Embedding Widget for Gtk+
+Version: %MOZILLA_VERSION%
+Requires: libxul-unstable = %MOZILLA_VERSION%
--- /dev/null	2008-01-08 09:03:16.107043921 +0100
+++ mozilla/xulrunner/installer/mozilla-gtkmozembed-embedding.pc.in	2008-01-08 09:44:56.000000000 +0100
@@ -0,0 +1,8 @@
+prefix=%prefix%
+sdkdir=%sdkdir%
+includedir=%includedir%
+
+Name: mozilla-gtkembedmoz
+Description: Static version-independent Mozilla Embedding Widget for Gtk+
+Version: %MOZILLA_VERSION%
+Requires: libxul-embedding-unstable = %MOZILLA_VERSION%
