https://bugzilla.mozilla.org/show_bug.cgi?id=457358

Index: toolkit/components/passwordmgr/src/storage-Legacy.js
===================================================================
RCS file: /cvsroot/mozilla/toolkit/components/passwordmgr/src/storage-Legacy.js,v
retrieving revision 1.29
diff -u -p -8 -r1.29 storage-Legacy.js
--- toolkit/components/passwordmgr/src/storage-Legacy.js	24 Sep 2008 17:39:34 -0000	1.29
+++ toolkit/components/passwordmgr/src/storage-Legacy.js	3 Oct 2008 21:00:37 -0000
@@ -79,16 +79,20 @@ LoginManagerStorage_legacy.prototype = {
         if (!this.__utfConverter) {
             this.__utfConverter = Cc["@mozilla.org/intl/scriptableunicodeconverter"].
                                   createInstance(Ci.nsIScriptableUnicodeConverter);
             this.__utfConverter.charset = "UTF-8";
         }
         return this.__utfConverter;
     },
 
+    _utfConverterReset : function() {
+        this.__utfConverter = null;
+    },
+
     __profileDir: null,  // nsIFile for the user's profile dir
     get _profileDir() {
         if (!this.__profileDir) {
             var dirService = Cc["@mozilla.org/file/directory_service;1"].
                              getService(Ci.nsIProperties);
             this.__profileDir = dirService.get("ProfD", Ci.nsIFile);
         }
         return this.__profileDir;
@@ -849,16 +853,17 @@ LoginManagerStorage_legacy.prototype = {
         var processEntry = false;
 
         do {
             var hasMore = lineStream.readLine(line);
             try {
               line.value = this._utfConverter.ConvertToUnicode(line.value);
             } catch (e) {
               this.log("Bad UTF8 conversion: " + line.value);
+              this._utfConverterReset();
             }
 
             switch (parseState) {
                 // Check file header
                 case STATE.HEADER:
                     if (line.value == "#2c") {
                         formatVersion = 0x2c;
                     } else if (line.value == "#2d") {
@@ -1290,16 +1295,19 @@ LoginManagerStorage_legacy.prototype = {
             } else {
                 plainOctet = this._decoderRing.decryptString(cipherText);
             }
             plainText = this._utfConverter.ConvertToUnicode(plainOctet);
         } catch (e) {
             this.log("Failed to decrypt string: " + cipherText +
                 " (" + e.name + ")");
 
+            // In the unlikely event the converter threw, reset it.
+            this._utfConverterReset();
+
             // If the user clicks Cancel, we get NS_ERROR_NOT_AVAILABLE.
             // If the cipherText is bad / wrong key, we get NS_ERROR_FAILURE
             // Wrong passwords are handled by the decoderRing reprompting;
             // we get no notification.
             if (e.result == Components.results.NS_ERROR_NOT_AVAILABLE)
                 userCanceled = true;
         }
 
Index: toolkit/components/passwordmgr/test/unit/test_storage_legacy_3.js
===================================================================
RCS file: /cvsroot/mozilla/toolkit/components/passwordmgr/test/unit/test_storage_legacy_3.js,v
retrieving revision 1.8
diff -u -p -8 -r1.8 test_storage_legacy_3.js
--- toolkit/components/passwordmgr/test/unit/test_storage_legacy_3.js	24 Sep 2008 17:39:34 -0000	1.8
+++ toolkit/components/passwordmgr/test/unit/test_storage_legacy_3.js	3 Oct 2008 21:00:37 -0000
@@ -565,16 +565,173 @@ LoginTest.checkStorageData(storage, [], 
 // The output file should contain valid UTF8 now, but the resulting
 // JS string value remains the same.
 
 testdesc = "[flush and reload for verification]"
 LoginTest.initStorage(storage, OUTDIR, "output-454708.txt");
 LoginTest.checkStorageData(storage, [], [bad8User]);
 
 
+/*
+ * ---------------------- Bug 457358 ----------------------
+ * need to reset UTF8 converter after it encounters invalid input
+ * (pwmgr problems in FF3.0.3)
+ */
+
+/* ========== 16 ========== */
+testnum++;
+testdesc = "ensure UTF8 converter isn't left in bad state."
+
+var utfRealm = "Acc" +
+               String.fromCharCode(0xe8) +
+               "s reserv" +
+               String.fromCharCode(0xe9);
+bad8User.init("https://bugzilla.mozilla.org", null, utfRealm,
+            "dummydude", "itsasecret", "", "");
+
+LoginTest.initStorage(storage,
+                      INDIR, "signons-457358-1.txt",
+                      OUTDIR, "output-457358-1.txt");
+LoginTest.checkStorageData(storage, [], [bad8User]);
+
+// The output file should contain valid UTF8 now, but the resulting
+// JS string value remains the same.
+
+testdesc = "[flush and reload for verification]"
+LoginTest.initStorage(storage, OUTDIR, "output-457358-1.txt");
+LoginTest.checkStorageData(storage, [], [bad8User]);
+
+/* ========== 17 ========== */
+testnum++;
+testdesc = "ensure UTF8 converter isn't left in bad state."
+
+// The username field here is "Acc" + String.fromCharCode(0xe8), but the last
+// character is invalid UTF8 -- it's the beginning of a multibyte sequence,
+// but at the end of the input. The unicode converter silently truncates the
+// string, and will throw when we feed it the next chunk of input (the
+// encrypted username)
+//
+// Test for an expected login -- everthing fine except for the truncated field
+// name (which we don't use anyway)
+bad8User.init("https://bugzilla.mozilla.org", "https://bugzilla.mozilla.org", null,
+            "dummydude", "itsasecret", "Acc", "pass");
+
+LoginTest.initStorage(storage,
+                      INDIR, "signons-457358-2.txt",
+                      OUTDIR, "output-457358-2.txt");
+LoginTest.checkStorageData(storage, [], [bad8User]);
+
+// The output file should contain valid UTF8 now, but the resulting
+// JS string value remains the same.
+
+testdesc = "[flush and reload for verification]"
+LoginTest.initStorage(storage, OUTDIR, "output-457358-2.txt");
+LoginTest.checkStorageData(storage, [], [bad8User]);
+
+/* ========== 18 ========== */
+testnum++;
+testdesc = "ensure UTF8 converter isn't left in bad state."
+
+// As with previous test, but triggered with both field names.
+
+bad8User.init("https://bugzilla.mozilla.org", "https://bugzilla.mozilla.org", null,
+            "dummydude", "itsasecret", "u-Acc", "p-Acc");
+
+LoginTest.initStorage(storage,
+                      INDIR, "signons-457358-3.txt",
+                      OUTDIR, "output-457358-3.txt");
+LoginTest.checkStorageData(storage, [], [bad8User]);
+
+// The output file should contain valid UTF8 now, but the resulting
+// JS string value remains the same.
+
+testdesc = "[flush and reload for verification]"
+LoginTest.initStorage(storage, OUTDIR, "output-457358-3.txt");
+LoginTest.checkStorageData(storage, [], [bad8User]);
+
+/* ========== 19 ========== */
+testnum++;
+testdesc = "ensure UTF8 converter isn't left in bad state."
+
+// Last character in the realm, this time. Not trunated, because the
+// "http://site.com (realm)" format means there's always a trailing character,
+// so the conversino throws.
+
+bad8User.init("https://bugzilla.mozilla.org", null, "Acc" + String.fromCharCode(0xe8),
+            "dummydude", "itsasecret", "", "");
+
+LoginTest.initStorage(storage,
+                      INDIR, "signons-457358-4.txt",
+                      OUTDIR, "output-457358-4.txt");
+LoginTest.checkStorageData(storage, [], [bad8User]);
+
+// The output file should contain valid UTF8 now, but the resulting
+// JS string value remains the same.
+
+testdesc = "[flush and reload for verification]"
+LoginTest.initStorage(storage, OUTDIR, "output-457358-4.txt");
+LoginTest.checkStorageData(storage, [], [bad8User]);
+
+/* ========== 20 ========== */
+testnum++;
+testdesc = "ensure UTF8 converter isn't left in bad state."
+
+// Like last test, but try adding and removing a login too.
+
+bad8User.init("https://bugzilla.mozilla.org", null, "Acc" + String.fromCharCode(0xe8),
+            "dummydude", "itsasecret", "", "");
+
+LoginTest.initStorage(storage,
+                      INDIR, "signons-457358-4.txt",
+                      OUTDIR, "output-457358-4b.txt");
+
+var anotherUser = Cc["@mozilla.org/login-manager/loginInfo;1"].
+              createInstance(Ci.nsILoginInfo);
+anotherUser.init("http://mozilla.org", null,
+                 String.fromCharCode(0xe8) + "xtra user " + String.fromCharCode(0x0163) + "est",
+                 "dummydude", "itsasecret", "", "");
+
+storage.addLogin(anotherUser);
+LoginTest.checkStorageData(storage, [], [bad8User, anotherUser]);
+
+testdesc = "[flush and reload for verification]"
+LoginTest.initStorage(storage, OUTDIR, "output-457358-4b.txt");
+LoginTest.checkStorageData(storage, [], [bad8User, anotherUser]);
+
+storage.removeLogin(anotherUser);
+LoginTest.checkStorageData(storage, [], [bad8User]);
+
+testdesc = "[flush and reload for verification 2]"
+LoginTest.initStorage(storage, OUTDIR, "output-457358-4b.txt");
+LoginTest.checkStorageData(storage, [], [bad8User]);
+
+/* ========== 21 ========== */
+testnum++;
+testdesc = "ensure UTF8 converter isn't left in bad state."
+
+// The first login's username (plaintext) is invalid UTF8. It's handled as a
+// bad decryption so we don't see the login, but make sure the other login in
+// the file is ok.
+
+// This is the first login:
+//bad8User.init("https://www.google.com", "https://www.google.com", null,
+//              "Acc" + String.fromCharCode(0xe8) + "ss", "passw", "un", "pw");
+anotherUser.init("https://bugzilla.mozilla.org", null, "extra user test",
+                 "dummydude", "itsasecret", "", "");
+
+LoginTest.initStorage(storage,
+                      INDIR, "signons-457358-5.txt",
+                      OUTDIR, "output-457358-5.txt");
+LoginTest.checkStorageData(storage, [], [anotherUser]);
+
+testdesc = "[flush and reload for verification]"
+LoginTest.initStorage(storage, OUTDIR, "output-457358-5.txt");
+LoginTest.checkStorageData(storage, [], [anotherUser]);
+
+
 
 
 /* ========== end ========== */
 } catch (e) {
     throw ("FAILED in test #" + testnum + " -- " + testdesc + ": " + e);
 }
 
 };
Index: toolkit/components/passwordmgr/test/unit/data/signons-457358-1.txt
===================================================================
RCS file: toolkit/components/passwordmgr/test/unit/data/signons-457358-1.txt
diff -N toolkit/components/passwordmgr/test/unit/data/signons-457358-1.txt
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ toolkit/components/passwordmgr/test/unit/data/signons-457358-1.txt	3 Oct 2008 21:00:37 -0000
@@ -0,0 +1,10 @@
+#2e
+.
+https://bugzilla.mozilla.org (Accès reservé)
+
+MDoEEPgAAAAAAAAAAAAAAAAAAAEwFAYIKoZIhvcNAwcECEnlbhAkNBbBBBCexD5eaffSLGH/ORiFlQ4X
+*
+MDoEEPgAAAAAAAAAAAAAAAAAAAEwFAYIKoZIhvcNAwcECHmiTaseYjkkBBAA0ILJTFSa5CnlpD5PTEYR
+
+---
+.
Index: toolkit/components/passwordmgr/test/unit/data/signons-457358-2.txt
===================================================================
RCS file: toolkit/components/passwordmgr/test/unit/data/signons-457358-2.txt
diff -N toolkit/components/passwordmgr/test/unit/data/signons-457358-2.txt
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ toolkit/components/passwordmgr/test/unit/data/signons-457358-2.txt	3 Oct 2008 21:00:37 -0000
@@ -0,0 +1,10 @@
+#2e
+.
+https://bugzilla.mozilla.org
+Accè
+MDoEEPgAAAAAAAAAAAAAAAAAAAEwFAYIKoZIhvcNAwcECEnlbhAkNBbBBBCexD5eaffSLGH/ORiFlQ4X
+*pass
+MDoEEPgAAAAAAAAAAAAAAAAAAAEwFAYIKoZIhvcNAwcECHmiTaseYjkkBBAA0ILJTFSa5CnlpD5PTEYR
+https://bugzilla.mozilla.org
+---
+.
Index: toolkit/components/passwordmgr/test/unit/data/signons-457358-3.txt
===================================================================
RCS file: toolkit/components/passwordmgr/test/unit/data/signons-457358-3.txt
diff -N toolkit/components/passwordmgr/test/unit/data/signons-457358-3.txt
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ toolkit/components/passwordmgr/test/unit/data/signons-457358-3.txt	3 Oct 2008 21:00:37 -0000
@@ -0,0 +1,10 @@
+#2e
+.
+https://bugzilla.mozilla.org
+u-Accè
+MDoEEPgAAAAAAAAAAAAAAAAAAAEwFAYIKoZIhvcNAwcECEnlbhAkNBbBBBCexD5eaffSLGH/ORiFlQ4X
+*p-Accè
+MDoEEPgAAAAAAAAAAAAAAAAAAAEwFAYIKoZIhvcNAwcECHmiTaseYjkkBBAA0ILJTFSa5CnlpD5PTEYR
+https://bugzilla.mozilla.org
+---
+.
Index: toolkit/components/passwordmgr/test/unit/data/signons-457358-4.txt
===================================================================
RCS file: toolkit/components/passwordmgr/test/unit/data/signons-457358-4.txt
diff -N toolkit/components/passwordmgr/test/unit/data/signons-457358-4.txt
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ toolkit/components/passwordmgr/test/unit/data/signons-457358-4.txt	3 Oct 2008 21:00:37 -0000
@@ -0,0 +1,10 @@
+#2e
+.
+https://bugzilla.mozilla.org (Accè)
+
+MDoEEPgAAAAAAAAAAAAAAAAAAAEwFAYIKoZIhvcNAwcECEnlbhAkNBbBBBCexD5eaffSLGH/ORiFlQ4X
+*
+MDoEEPgAAAAAAAAAAAAAAAAAAAEwFAYIKoZIhvcNAwcECHmiTaseYjkkBBAA0ILJTFSa5CnlpD5PTEYR
+
+---
+.
Index: toolkit/components/passwordmgr/test/unit/data/signons-457358-5.txt
===================================================================
RCS file: toolkit/components/passwordmgr/test/unit/data/signons-457358-5.txt
diff -N toolkit/components/passwordmgr/test/unit/data/signons-457358-5.txt
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ toolkit/components/passwordmgr/test/unit/data/signons-457358-5.txt	3 Oct 2008 21:00:37 -0000
@@ -0,0 +1,18 @@
+#2e
+.
+https://www.google.com
+un
+MDIEEPgAAAAAAAAAAAAAAAAAAAEwFAYIKoZIhvcNAwcECAcH36OqRDPBBAj4u1kKe2oefQ==
+*pw
+MDIEEPgAAAAAAAAAAAAAAAAAAAEwFAYIKoZIhvcNAwcECDG4MxIBIm3IBAjLrxoeWcW5eg==
+https://www.google.com
+---
+.
+https://bugzilla.mozilla.org (extra user test)
+
+MDoEEPgAAAAAAAAAAAAAAAAAAAEwFAYIKoZIhvcNAwcECEnlbhAkNBbBBBCexD5eaffSLGH/ORiFlQ4X
+*
+MDoEEPgAAAAAAAAAAAAAAAAAAAEwFAYIKoZIhvcNAwcECHmiTaseYjkkBBAA0ILJTFSa5CnlpD5PTEYR
+
+---
+.
