From: timeless@mozdev.org

Bug 499295 NULL crash [@ nsPluginInstancePeerImpl::GetJSContext]
r=jst sr=jst

diff --git a/modules/plugin/base/src/nsPluginInstancePeer.cpp b/modules/plugin/base/src/nsPluginInstancePeer.cpp
--- a/modules/plugin/base/src/nsPluginInstancePeer.cpp
+++ b/modules/plugin/base/src/nsPluginInstancePeer.cpp
@@ -689,8 +689,10 @@ nsPluginInstancePeerImpl::GetJSContext(J
 {
   *outContext = NULL;
   nsresult rv = NS_ERROR_FAILURE;
+  if (!mOwner)
+    return rv;
+
   nsCOMPtr<nsIDocument> document;
-
   rv = mOwner->GetDocument(getter_AddRefs(document));
 
   if (NS_SUCCEEDED(rv) && document) {
