diff -up xulrunner-2.0/mozilla-central/js/src/jsval.h.64bit-big-endian xulrunner-2.0/mozilla-central/js/src/jsval.h
--- xulrunner-2.0/mozilla-central/js/src/jsval.h.64bit-big-endian	2010-12-15 02:03:06.000000000 +0100
+++ xulrunner-2.0/mozilla-central/js/src/jsval.h	2010-12-22 12:00:57.000000000 +0100
@@ -334,6 +334,32 @@ typedef union jsval_layout
     double asDouble;
     void *asPtr;
 } jsval_layout;
+# elif JS_BITS_PER_WORD == 64
+typedef union jsval_layout
+{
+    uint64 asBits;
+#ifndef _WIN64
+    /* MSVC does not pack these correctly :-( */
+    struct {
+        uint64             payload47 : 47;
+        JSValueTag         tag : 17;
+    } debugView;
+#endif
+    struct {
+        union {
+            int32          i32;
+            uint32         u32;
+            JSBool         boo;
+            JSString       *str;
+            JSObject       *obj;
+            void           *ptr;
+            JSWhyMagic     why;
+            jsuword        word;
+        } payload;
+    } s;
+    double asDouble;
+    void *asPtr;
+} jsval_layout;
 # endif /* JS_BITS_PER_WORD */
 #endif  /* defined(IS_LITTLE_ENDIAN) */
 
