--- widget/src/gtk2/nsLookAndFeel.cpp.old	2007-11-29 14:21:09.000000000 +0100
+++ widget/src/gtk2/nsLookAndFeel.cpp	2007-11-29 14:22:37.000000000 +0100
@@ -83,6 +83,7 @@
 
 nsresult nsLookAndFeel::NativeGetColor(const nsColorID aID, nscolor& aColor)
 {
+#if 0
     nsresult res = NS_OK;
 
     switch (aID) {
@@ -308,6 +309,8 @@
     }
 
     return res;
+#endif
+    return NS_ERROR_FAILURE;
 }
 
 static PRInt32 CheckWidgetStyle(GtkWidget* aWidget, const char* aStyle, PRInt32 aMetric) {
--- widget/src/gtk2/Makefile.in.no-native-theme	2007-10-06 14:08:46.000000000 +0200
+++ widget/src/gtk2/Makefile.in	2007-12-01 16:04:12.000000000 +0100
@@ -52,8 +52,6 @@
 
 PACKAGE_FILE = widget_gtk2.pkg
 
-NATIVE_THEME_SUPPORT = 1
-
 REQUIRES	= xpcom \
 		  string \
 		  gfx \
@@ -139,8 +137,9 @@
 		mozcontainer.h \
 		$(NULL)
 
-ifdef NATIVE_THEME_SUPPORT
 CSRCS		+= gtk2drawing.c
+
+ifdef NATIVE_THEME_SUPPORT
 CPPSRCS		+= nsNativeThemeGTK.cpp
 DEFINES		+= -DNATIVE_THEME_SUPPORT
 endif
