Index: config/rules.mk
===================================================================
RCS file: /cvsroot/mozilla/config/rules.mk,v
retrieving revision 3.554
diff -u -r3.554 rules.mk
--- config/rules.mk	14 May 2007 14:21:44 -0000	3.554
+++ config/rules.mk	21 May 2007 00:41:20 -0000
@@ -775,6 +775,12 @@
 install:: $(SDK_HEADERS)
 	$(SYSINSTALL) $(IFLAGS1) $^ $(DESTDIR)$(includedir)/$(MODULE)
 endif
+
+ifneq (,$(SDK_LIBRARY))
+install:: $(SDK_LIBRARY)
+	$(SYSINSTALL) $(IFLAGS1) $^ $(DESTDIR)$(mozappdir)
+endif
+
 endif
 
 install:: $(SHARED_LIBRARY) $(IMPORT_LIBRARY) $(LIBRARY) $(PROGRAM) $(SIMPLE_PROGRAMS) $(JAVA_LIBRARY)
--- extensions/python/xpcom/src/module/Makefile.in.orig	2007-05-28 20:48:26.000000000 +0200
+++ extensions/python/xpcom/src/module/Makefile.in	2007-05-28 20:49:03.000000000 +0200
@@ -95,8 +95,8 @@
 
 install::
 ifneq ($(OS_ARCH),WINNT)
-	$(SYSINSTALL) $(IFLAGS2) $(SHARED_LIBRARY) $(DISTDIR)$(pyexecdir)/xpcom
-	mv $(DISTDIR)$(pyexecdir)/xpcom/$(SHARED_LIBRARY) $(DISTDIR)$(pyexecdir)/xpcom/_xpcom$(MOZ_PYTHON_DLL_SUFFIX)
+	$(SYSINSTALL) $(IFLAGS2) $(SHARED_LIBRARY) $(DESTDIR)$(pyexecdir)/xpcom
+	mv $(DESTDIR)$(pyexecdir)/xpcom/$(SHARED_LIBRARY) $(DESTDIR)$(pyexecdir)/xpcom/_xpcom$(MOZ_PYTHON_DLL_SUFFIX)
 endif
 
 clobber::
--- extensions/python/xpcom/Makefile.in.orig	2007-05-29 10:17:10.000000000 +0200
+++ extensions/python/xpcom/Makefile.in	2007-05-29 10:17:22.000000000 +0200
@@ -97,9 +97,9 @@
 	$(INSTALL) $(PYSRCS_XPCOMTOOLS) $(DIST)/bin/python/xpcom/tools
 
 install::
-	$(SYSINSTALL) $(IFLAGS1) $(PYSRCS_XPCOM) $(DISTDIR)$(pyexecdir)/xpcom
-	$(SYSINSTALL) $(IFLAGS1) $(PYSRCS_XPCOMCLIENT) $(DISTDIR)$(pyexecdir)/xpcom/client
-	$(SYSINSTALL) $(IFLAGS1) $(PYSRCS_XPCOMSERVER) $(DISTDIR)$(pyexecdir)/xpcom/server
+	$(SYSINSTALL) $(IFLAGS1) $(PYSRCS_XPCOM) $(DESTDIR)$(pyexecdir)/xpcom
+	$(SYSINSTALL) $(IFLAGS1) $(PYSRCS_XPCOMCLIENT) $(DESTDIR)$(pyexecdir)/xpcom/client
+	$(SYSINSTALL) $(IFLAGS1) $(PYSRCS_XPCOMSERVER) $(DESTDIR)$(pyexecdir)/xpcom/server
 
 clobber::
 	rm -rf $(DIST)/bin/python/xpcom
Index: extensions/python/xpcom/src/PyIInterfaceInfo.cpp
===================================================================
RCS file: /cvsroot/mozilla/extensions/python/xpcom/src/PyIInterfaceInfo.cpp,v
retrieving revision 1.14
diff -u -r1.14 PyIInterfaceInfo.cpp
--- extensions/python/xpcom/src/PyIInterfaceInfo.cpp	12 Jan 2008 04:30:44 -0000	1.14
+++ extensions/python/xpcom/src/PyIInterfaceInfo.cpp	30 Jan 2008 13:58:59 -0000
@@ -88,7 +88,7 @@
 	if (pI==NULL)
 		return NULL;
 
-	nsIID *iid_ret;
+	const nsIID *iid_ret;
 	nsresult r;
 	Py_BEGIN_ALLOW_THREADS;
 	r = pI->GetIIDShared(&iid_ret);
