diff -up mozilla/toolkit/xre/Makefile.in.old mozilla/toolkit/xre/Makefile.in
--- mozilla/toolkit/xre/Makefile.in.old	2007-09-17 17:13:01.000000000 +0200
+++ mozilla/toolkit/xre/Makefile.in	2007-09-17 17:12:33.000000000 +0200
@@ -263,5 +263,5 @@ platform.ini: FORCE
 libs:: platform.ini
 	$(INSTALL) $^ $(DIST)/bin
 
-install::
-	$(INSTALL) $(IFLAGS1) $^ $(DESTDIR)$(mozappdir)
+install:: platform.ini
+	$(SYSINSTALL) $(IFLAGS1) $^ $(DESTDIR)$(mozappdir)
