--- widget/src/gtk2/Makefile.in.old	2007-11-29 14:19:58.000000000 +0100
+++ widget/src/gtk2/Makefile.in	2007-11-29 14:20:03.000000000 +0100
@@ -52,7 +52,7 @@
 
 PACKAGE_FILE = widget_gtk2.pkg
 
-NATIVE_THEME_SUPPORT = 1
+NATIVE_THEME_SUPPORT = 0
 
 REQUIRES	= xpcom \
 		  string \
--- widget/src/gtk2/nsLookAndFeel.cpp.old	2007-11-29 14:21:09.000000000 +0100
+++ widget/src/gtk2/nsLookAndFeel.cpp	2007-11-29 14:22:37.000000000 +0100
@@ -83,6 +83,7 @@
 
 nsresult nsLookAndFeel::NativeGetColor(const nsColorID aID, nscolor& aColor)
 {
+#if 0
     nsresult res = NS_OK;
 
     switch (aID) {
@@ -308,6 +309,8 @@
     }
 
     return res;
+#endif
+    return NS_ERROR_FAILURE;
 }
 
 static PRInt32 CheckWidgetStyle(GtkWidget* aWidget, const char* aStyle, PRInt32 aMetric) {
